web/src/pages/Route/transform.ts

```diff
@@ -82,6 +82,7 @@ export const transformStepData = ({
       !Object.keys(step3DataCloned.script || {}).length ? 'script' : '',
       form1Data.hosts.filter(Boolean).length === 0 ? 'hosts' : '',
       form1Data.redirectOption === 'disabled' ? 'redirect' : '',
+      data.remote_addrs?.filter(Boolean).length === 0 ? 'remote_addrs' : '',
```

Comment:
Very ugly and not easy to maintain on the frontend's side, I agree with no matter empty array or omit that field should be valid for the backend.

cc @membphis @liuxiran @moonming @ShiningRush @tokers

Reply:
@juzhiyuan you can create a new issue at APISIX, we can discuss this first.

we can use a better way in milestone `2.2`.

Reply:
fot the data validation, I agree with  @juzhiyuan , this will also be flexible for people who use api.

If Fe must comply with omit field which is not configed, we may choose more elegant way to implement this transform, e.g: 
write a until func to handle omit params.

Reply:
Those ugly fallbacks would be dropped according to https://github.com/apache/apisix/issues/2924 , we use current way to be compatible with APISIX 2.1.

Reply:
Why `empty array` is equal to `omit that field`? They are quite different things. The former one is a `[]`, while the latter one is `nil`.

Reply:
They are different indeed, but the frontend doesn't care about which array-like parameters should be omitted before sending request as well.

Reply:
> the frontend doesn't care about which array-like parameters should be omitted before sending requests

Yes, but the backends care. This change just shift the duty to the backend.
There are three problems:
1. We have two backends. It is hard to sync the validation.
2. jsonschema treat `empty array` and `omit array field` different. We have to write code manually twice.
3. What if one day we want to distinguish `[]` from `nil`?

Reply:
resolved and discussing here[1]

[1] https://github.com/apache/apisix/issues/2924#issuecomment-737037777